pkg/detectors/azuredevopspersonalaccesstoken/azuredevopspersonalaccesstoken.go

```diff
@@ -0,0 +1,98 @@
+package azuredevopspersonalaccesstoken
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"azure"}) + `\b([0-9a-z]{52})\b`)
+	orgPat = regexp.MustCompile(detectors.PrefixRegex([]string{"azure"}) + `\b([0-9a-z]{7,40})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"azure"}
+}
+
+// FromData will find and optionally verify AzureDevopsPersonalAccessToken secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	orgMatches := orgPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+		for _, orgMatch := range orgMatches {
+			if len(orgMatch) != 2 {
+				continue
+			}
+			resOrgMatch := strings.TrimSpace(orgMatch[1])
+
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_AzureDevopsPersonalAccessToken,
+				Raw:          []byte(resMatch),
```

Comment:
Since this is a multi-part credential (org and token), we need to have both values included for deduplication. I'll add those now and merge.